tests/NSubstitute.Acceptance.Specs/EventRaising.cs

```diff
@@ -340,4 +354,29 @@ public class CustomEventArgsWithNoDefaultCtor : EventArgs
     {
         public CustomEventArgsWithNoDefaultCtor(string arg) { }
     }
+
+    public class CustomEventArgsWithInternalDefaultConstructor : EventArgs
+    {
+        internal CustomEventArgsWithInternalDefaultConstructor() { }
```

Comment:
> Ultimately it may well be the user's responsibility to be aware of their use cases in this regard, rather than the library restricting potentially legitimate use cases, like the private constructor.
>
> Please let me know what you think of this.

I do have some reservations about this. It's easy for users to make decisions based on current state, then another project member to make a change expecting the private constructor would not be automatically called. Maybe responsibility for creating non-creatable instances would be best in other code or another library?

```csharp
// CreateAny in library or project code:
sub.MyEvent += Raise.EventWith<CustomEventArgsWithInternalDefaultConstructor>(this, CreateAny());

// Singleton example
sub.MyEvent += Raise.EventWith<CustomEventArgsWithInternalDefaultConstructor>(this, CustomEventArgsWithInternalDefaultConstructor.GetInstance());
```

NSubstitute has generally tried to play it safe with new-ing up stuff (see [warning about pure virtual classes](https://nsubstitute.github.io/help/creating-a-substitute/#substituting-infrequently-and-carefully-for-classes)).

If you have a strong motivating example for this then maybe we could add an overload that opts-in to new'ing up inaccessible ctors?

Reply:
Thank you for the clarification, @dtchepak!

I have settled on the internal-only default constructor solution, and also added negative test for private default constructor.